platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/ResolvedArtifactCollectingVisitor.java

```diff
@@ -28,14 +28,15 @@
 import org.gradle.internal.DisplayName;
 
 import java.io.File;
-import java.util.HashSet;
+import java.util.LinkedHashMap;
 import java.util.List;
+import java.util.Map;
+import java.util.Objects;
 import java.util.Set;
 
 public class ResolvedArtifactCollectingVisitor implements ArtifactVisitor {
-    private final Set<ResolvedArtifactResult> artifacts = Sets.newLinkedHashSet();
+    private final Map<ArtifactIdentifier, DefaultResolvedArtifactResult> artifacts = new LinkedHashMap<>();
```

Comment:
💅 Could use `Maps.newLinkedHashMap()`, just to be consistent with the other field in this class.

Reply:
Or preferably move away from `Maps.` and `Lists.` instead of continuing to use these methods. Why do they even exist? If I am missing some good reason then that's one thing but I don't see why we'd use these simple wrappers over the normal constructors

Reply:
So TIL on these methods' javadoc it now says:

> Note: this method is now unnecessary and should be treated as deprecated. Instead, use the LinkedHashSet constructor directly, taking advantage of "diamond" syntax .

We should probably remove all usages in favor of constructor calls and add a checkstyle rule to keep them out.

Reply:
I got this together with ArchUnit: https://github.com/gradle/gradle/pull/27244.